bookwyrm/static/js/bookwyrm.js

```diff
@@ -0,0 +1,308 @@
+        let hidden = trigger.closest('.hidden-form').querySelectorAll('.is-hidden')[0];
+
+        this.addRemoveClass(hidden, 'is-hidden', !hidden);
+    }
+
+    /**
+     * Execute actions on targets based on triggers.
+     *
+     * @param  {Event} event
+     * @return {undefined}
+     */
+    toggleAction(event) {
+        let trigger = event.currentTarget;
+        let pressed = trigger.getAttribute('aria-pressed') === 'false';
+        let targetId = trigger.dataset.controls;
+
+        // Toggle pressed status on all triggers controlling the same target.
+        document.querySelectorAll('[data-controls="' + targetId + '"]')
+            .forEach(otherTrigger => otherTrigger.setAttribute(
+                'aria-pressed',
+                otherTrigger.getAttribute('aria-pressed') === 'false'
+            ));
+
+        // @todo Find a better way to handle the exception.
+        if (targetId && ! trigger.classList.contains('pulldown-menu')) {
+            let target = document.getElementById(targetId);
+
+            this.addRemoveClass(target, 'is-hidden', !pressed);
+            this.addRemoveClass(target, 'is-active', pressed);
+        }
+
+        // Show/hide pulldown-menus.
+        if (trigger.classList.contains('pulldown-menu')) {
+            this.toggleMenu(trigger, targetId);
+        }
+
+        // Show/hide container.
+        let container = document.getElementById('hide-' + targetId);
+
+        if (container) {
+            this.toggleContainer(container, pressed);
+        }
+
+        // Check checkbox, if appropriate.
+        let checkbox = trigger.dataset['controls-checkbox'];
+
+        if (checkbox) {
+            this.toggleCheckbox(checkbox, pressed);
+        }
+
+        // Set focus, if appropriate.
+        let focus = trigger.dataset['focus-target'];
+
+        if (focus) {
+            this.toggleFocus(focus);
+        }
+    }
+
+    /**
+     * Show or hide menus.
+     *
+     * @param  {Event} event
+     * @return {undefined}
+     */
+    toggleMenu(trigger, targetId) {
+        let expanded = trigger.getAttribute('aria-expanded') == 'false';
+
+        trigger.setAttribute('aria-expanded', expanded);
+
+        if (targetId) {
+            let target = document.getElementById(targetId);
+
+            this.addRemoveClass(target, 'is-active', expanded);
+        }
+    }
+
+    /**
+     * Show or hide generic containers.
+     *
+     * @param  {object}  container - DOM node
+     * @param  {boolean} pressed   - Is the trigger pressed?
+     * @return {undefined}
+     */
+    toggleContainer(container, pressed) {
+        this.addRemoveClass(container, 'is-hidden', pressed);
+    }
+
+    /**
+     * Check or uncheck a checbox.
+     *
+     * @param  {object}  checkbox - DOM node
+     * @param  {boolean} pressed  - Is the trigger pressed?
+     * @return {undefined}
+     */
+    toggleCheckbox(checkbox, pressed) {
+        document.getElementById(checkbox).checked = !!pressed;
```

Comment:
this doesn't seem to be doing its job -- the checkbox isn't being toggled, which causes statuses to post without their content warnings